feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/domain/referendum/tindergov/RealTinderGovInteractor.kt

```diff
@@ -69,4 +108,28 @@ class RealTinderGovInteractor(
             null
         }
     }
+
+    override suspend fun setVotingPower(chainId: ChainId, amount: BigInteger, conviction: Conviction) {
+        val metaAccount = accountRepository.getSelectedMetaAccount()
+        tinderGovVotingPowerRepository.setVotingPower(VotingPower(metaAccount.id, chainId, amount, conviction))
+    }
+
+    override suspend fun getVotingPower(metaId: Long, chainId: ChainId): VotingPower? {
+        return tinderGovVotingPowerRepository.getVotingPower(metaId, chainId)
+    }
+
+    override suspend fun isVotingPowerAvailable(): Boolean {
+        val metaAccount = accountRepository.getSelectedMetaAccount()
+        val chain = governanceSharedState.chain()
+        return getVotingPower(metaAccount.id, chain.id) != null
+    }
+
+    override suspend fun isSufficientAmountToVote(): Boolean {
+        val metaAccount = accountRepository.getSelectedMetaAccount()
+        val chain = governanceSharedState.chain()
+        val votingPower = getVotingPower(metaAccount.id, chain.id) ?: return false
+        val asset = walletRepository.getAsset(metaAccount.id, chain.utilityAsset) ?: return false
+
+        return asset.totalInPlanks >= votingPower.amount
```

Comment:
We're currently using `asset.free` as a maximum amount for vote though `total` is the correct one. I suggest to make it `free` to be consistent with other flows for now